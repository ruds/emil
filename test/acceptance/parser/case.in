case {foo=waldo, bar=baz 2} : quux
 of {foo=2, bar=5} => 23
  | {foo=3, ...} => (+) 42 1
  | _ => -1;
