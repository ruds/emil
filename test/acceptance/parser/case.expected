0001 (ExprTopDecl
         :expr
         (CaseExpr
             :expr
             (TypedExpr
                 :expr
                 (RecordExpr
                     :rows
                     (
                         (RecRowExpr
                             :label
                             "666F6F"
                             :val
                             (IdentifierExpr
                                 :qualifiers
                                 ()
                                 :identifier
                                 "77616C646F"
                                 :is_op
                                 false
                                 :is_prefix_op
                                 false))
                         (RecRowExpr
                             :label
                             "626172"
                             :val
                             (ApplicationExpr
                                 :exprs
                                 (
                                     (IdentifierExpr
                                         :qualifiers
                                         ()
                                         :identifier
                                         "62617A"
                                         :is_op
                                         false
                                         :is_prefix_op
                                         false)
                                     (BigintLiteralExpr
                                         :val
                                         2)
                                 )))
                     ))
                 :type
                 (TyconTypeExpr
                     :qualifiers
                     ()
                     :identifier
                     "71757578"
                     :types
                     ()))
             :cases
             (
                 ((RecordPattern
                     :rows
                     (
                         (RecRowPattern
                             :label
                             "666F6F"
                             :pattern
                             (LiteralPattern
                                 :val
                                 (BigintLiteralExpr
                                     :val
                                     2)))
                         (RecRowPattern
                             :label
                             "626172"
                             :pattern
                             (LiteralPattern
                                 :val
                                 (BigintLiteralExpr
                                     :val
                                     5)))
                     ))
                   . (BigintLiteralExpr
                         :val
                         23))
                 ((PolyRecordPattern
                     :rows
                     ((RecRowPattern
                         :label
                         "666F6F"
                         :pattern
                         (LiteralPattern
                             :val
                             (BigintLiteralExpr
                                 :val
                                 3)))))
                   . (ApplicationExpr
                         :exprs
                         (
                             (IdentifierExpr
                                 :qualifiers
                                 ()
                                 :identifier
                                 "2B"
                                 :is_op
                                 true
                                 :is_prefix_op
                                 false)
                             (BigintLiteralExpr
                                 :val
                                 42)
                             (BigintLiteralExpr
                                 :val
                                 1)
                         )))
                 ((WildcardPattern)
                   . (BigintLiteralExpr
                         :val
                         -1))
             )))
0005 (EndOfFileTopDecl)
